charts/datadog/values.yaml

```
@@ -308,6 +308,20 @@ datadog:
     # datadog.apm.hostSocketPath -- Host path to the trace-agent socket
     hostSocketPath: /var/run/datadog/
 
+  # datadog.otlp -- OTLP ingest related configuration
+  ## Set OTLP/gRPC or OTLP/HTTP endpoint to enable OTLP ingest.
+  otlp: {}
+  #  receiver:
+  #    protocols:
+  #      grpc:
+  #        # datadog.otlp.receiver.procotols.grpc.endpoint -- OTLP/gRPC endpoint
+  #        ## Set explicitly to enable OTLP/gRPC for all supported datatypes.
+  #        # endpoint: "0.0.0.0:4317"
+  #      http:
+  #        # datadog.otlp.receiver.procotols.http.endpoint -- OTLP/HTTP endpoint
+  #        ## Set explicitly to enable OTLP/HTTP for all supported datatypes.
+  #        # endpoint: "0.0.0.0:4318"
```

IMO we should try to improve the user experience, by providing an easier configuration:
* a boolean to enable it.
* a default configuration.

```suggestion
  otlp: 
    enabled: false
    receiver:
      protocols:
        grpc:
          # datadog.otlp.receiver.procotols.grpc.endpoint -- OTLP/gRPC endpoint
          ## Set explicitly to enable OTLP/gRPC for all supported datatypes.
          # endpoint: "0.0.0.0:4317"
        http:
          # datadog.otlp.receiver.procotols.http.endpoint -- OTLP/HTTP endpoint
          ## Set explicitly to enable OTLP/HTTP for all supported datatypes.
          # endpoint: "0.0.0.0:4318"
```

---

I guess you would want something like this instead?

```
otlp:
  receiver:
    protocols:
      grpc:
        enabled: false
       http:
         enabled: false
```

Most people would use only one of the protocols for sending telemetry data. This makes sense, the way this works on every other OpenTelemetry product (Datadog or other companies) is to have:

```
otlp:
  receiver:
    protocols:
      grpc:
```

be the "use the default settings for gRPC", but I don't think Helm supports this (or at least, I haven't been able to make it work in any way), so enabled seems like the next best thing.

---

IMO if it makes sens we can enable both (grpc, http) by default. And if a customer request provide a way to disable it.


```yaml
otlp:
  enabled: false
  receiver:
    protocols:
      grpc:
        enable: true
       http:
         enable: true
```

or if `endpoint` is empty disable it

```yaml
otlp:
  enabled: false
  receiver:
    protocols:
      grpc:
        endpoint: ""
      http:
        endpoint: ""
```

From our experience, we should provide a configuration as simple as possible.

---

@clamoriniere I made it work in the same way it works in the Datadog Agent YAML and the OpenTelemetry Collector OTLP receiver configuration. That is, for example:

```
datadog:
  otlp:
     receiver:
       protocols:
          grpc:
```

Will enable the OTLP/gRPC endpoint. I feel like this is simpler (people don't need to set the endpoint) and consistent with what people coming from other vendors or migrating from local Datadog Agent YAML would expect.

Do you feel this is acceptable?